src/lib/modules/blockchain_process/proxy.js

```diff
@@ -1,156 +1,180 @@
-const httpProxy = require('http-proxy');
-const http = require('http');
-const constants = require('../../constants.json');
-const utils = require('../../utils/utils');
-
-let commList = {};
-let transactions = {};
-let receipts = {};
+/* global __ exports require */
 
+const Asm = require('stream-json/Assembler');
 const {canonicalHost, defaultHost} = require('../../utils/host');
+const {chain}  = require('stream-chain');
+const cloneable = require('cloneable-readable');
+const constants = require('../../constants.json');
+const express = require('express');
+const {parser} = require('stream-json');
+const proxyMiddleware = require('http-proxy-middleware');
+const pump = require('pump');
+const utils = require('../../utils/utils');
+const WebSocket = require('ws');
+const WsParser = require('simples/lib/parsers/ws');
 
-const parseRequest = function (reqBody) {
-  let jsonO;
+const parseJsonMaybe = (string) => {
+  let object;
   try {
-    jsonO = JSON.parse(reqBody);
+    if (string && typeof string === 'string') object = JSON.parse(string);
   } catch (e) {
-    return; // Request is not a json. Do nothing
-  }
-  if (jsonO.method === "eth_sendTransaction") {
-    commList[jsonO.id] = {
-      type: 'contract-log',
-      address: jsonO.params[0].to,
-      data: jsonO.params[0].data
-    };
-  } else if (jsonO.method === "eth_getTransactionReceipt") {
-    if (transactions[jsonO.params[0]]) {
-      transactions[jsonO.params[0]].receiptId = jsonO.id;
-      receipts[jsonO.id] = transactions[jsonO.params[0]].commListId;
-    }
+    console.error('Error parsing string as JSON', string);
+  } finally {
+    // eslint-disable-next-line no-unsafe-finally
+    return object;
   }
 };
 
-const parseResponse = function (ipc, resBody) {
-  let jsonO;
-  try {
-    jsonO = JSON.parse(resBody);
-  } catch (e) {
-    return; // Response is not a json. Do nothing
-  }
-  if (commList[jsonO.id]) {
-    commList[jsonO.id].transactionHash = jsonO.result;
-    transactions[jsonO.result] = {commListId: jsonO.id};
-  } else if (receipts[jsonO.id] && jsonO.result && jsonO.result.blockNumber) {
-    // TODO find out why commList[receipts[jsonO.id]] is sometimes not defined
-    if (!commList[receipts[jsonO.id]]) {
-      commList[receipts[jsonO.id]] = {};
+exports.serve = async (ipc, host, port, ws, origin) => {
+  const commList = {};
+  const receipts = {};
+  const transactions = {};
+
+  const trackRequest = (req) => {
+    if (!req) return;
+    try {
+      if (req.method === 'eth_sendTransaction') {
```

Comment:
Can we put this magic string in a constant?

Reply:
See: https://github.com/embark-framework/embark/pull/1129#discussion_r237183179